packages/autocomplete-js/src/render.tsx

```diff
@@ -90,87 +90,91 @@ export function renderPanel<TItem extends BaseItem>(
 
   dom.panel.classList.toggle('aa-Panel--stalled', state.status === 'stalled');
 
-  const sections = state.collections.map(({ source, items }, sourceIndex) => (
-    <section
-      key={sourceIndex}
-      className={classNames.source}
-      data-autocomplete-source-id={source.sourceId}
-    >
-      {source.templates.header && (
-        <div className={classNames.sourceHeader}>
-          {source.templates.header({
-            components,
-            createElement,
-            Fragment,
-            items,
-            source,
-            state,
-          })}
-        </div>
-      )}
+  const sections = state.collections
+    .filter(
+      ({ source, items }) => source.templates.noResults || items.length !== 0
```

Comment:
We use this comparison in the codebase:

```suggestion
      ({ source, items }) => source.templates.noResults || items.length > 0
```